### PR TITLE
fix up cli tabular display and locationRef

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4536

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by pandeyabs** [comment]
This has a few linter errors around generics/interface compliance. Could you please fix them? I can review it after.

**Comment by ryanfkeepers** [comment]
@pandeyabs should be good to go now

File: src/pkg/backup/details/groups.go [hunk 1]
```diff
@@ -121,6 +123,14 @@ func (i GroupsInfo) Values() []string {
 			dttm.FormatToTabularDisplay(i.Message.CreatedAt),
 			lastReply,
 		}
+	case GroupsConversationPost:
+		return []string{
+			// html parsing may produce newlines, which we'll want to avoid
+			strings.ReplaceAll(i.Post.Preview, "\n", "\\n"),
```
Comment (pandeyabs) [inline src/pkg/backup/details/groups.go:129]: I think details are broken for conversations, including this branch. 

I tried `./corso backup create groups --group CorsoCITeam --data conversations` 
`  Posts                                       2s done (found 10 conversations)`

followed by `./corso backup details groups  --backup   07a2b25b-495c-4f09-9366-d4361e7b32a2`

It returns `no items match the provided selectors`.

File: src/pkg/services/m365/api/conversations_test.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (suite *ConversationsAPIUnitSuite) TestConversationPostInfo() {
 						Preview:   "",
 						Size:      0,
 						// TODO: feed the subject in from the conversation
-						Subject: "",
+						Topic: "",
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/conversations_test.go:99]: Above TODO can be removed.

File: src/pkg/backup/details/groups.go (hunk 1 above)
Comment (pandeyabs) [reply src/pkg/backup/details/groups.go:129]: I'll debug it.

**Review by pandeyabs — COMMENTED** [review]
PR looks good overall. It also fixes the incremental backup failure I was seeing with main during merge details, `merging details: merging merge backup base details: getting updated info for entry: getting previous item location: unable to extract LocationRef from RepoRef`. 

 Details are still broken, I'll debug it in the next hour. I can approve this PR after what I find. We can push a fix separately. Should be fine since conversations backup is still in preview.

**Review by pandeyabs — APPROVED** [review]

File: src/pkg/backup/details/groups.go (hunk 1 above)
Comment (pandeyabs) [reply src/pkg/backup/details/groups.go:129]: @ryanfkeepers i have the details fix here - https://github.com/alcionai/corso/issues/4780 
Could you please merge it in this PR before pushing it?

Comment (pandeyabs) [reply src/pkg/backup/details/groups.go:129]: I've merged #4826 into this PR. @ryanfkeepers ok to merge this PR into main?

**Pull request merged**